evaluation_wrapper.py

```diff
@@ -0,0 +1,120 @@
+                                                    batch_size=DEFAULT_BATCH_SIZE,
+                                                    online_fetch=False,
+                                                    is_eval=True)
+        self.cocoGt = COCO(DEFAULT_VAL_ANNOT_PATH)
+        print("Initialized Evaluation Wrapper!")
+
+    def visualizeHeatmaps(self, genEnum=Generator.representative_set_gen):
+        self.visualize(genEnum=genEnum, visualize_heatmaps=True, visualize_scatter=False, visualize_skeleton=False)
+
+    def visualizeKeypoints(self, genEnum=Generator.representative_set_gen, visualize_skeleton=True):
+        self.visualize(genEnum=genEnum, visualize_heatmaps=False, visualize_scatter=True, visualize_skeleton=visualize_skeleton)
+
+    # Heatmaps is by default False because it is extremely processor intensive to calculate
+    def visualize(self, genEnum=Generator.representative_set_gen, visualize_heatmaps=False, visualize_scatter=True, visualize_skeleton=True):
+        gen = self._get_generator(genEnum)
+
+        for X_batch, y_stacked, m_batch in gen:
+            y_batch = y_stacked[0] # take first hourglass section
+
+            img_id_batch = [m['ann_id'] for m in m_batch] # image IDs are the annotation IDs
+
+            predicted_heatmaps_batch = self.eval.predict_heatmaps(X_batch)
+
+            if visualize_heatmaps:
+                self.eval.visualize_heatmaps(X_batch, y_batch, img_id_batch, predicted_heatmaps_batch)
+
+            if visualize_scatter or visualize_skeleton:
+                # Get predicted keypoints from last hourglass (last element of list)
+                # Dimensions are (hourglass_layer, batch, x, y, keypoint)
+                keypoints_batch = self.eval.heatmaps_to_keypoints_batch(predicted_heatmaps_batch)
+
+                if visualize_skeleton:
+                    # Plot only skeleton
+                    self.eval.visualize_keypoints(np.zeros(INPUT_DIM), keypoints_batch, [img_id + '_no_bg' for img_id in img_id_batch])
+
+                # Plot skeleton with image
+                self.eval.visualize_keypoints(X_batch, keypoints_batch, img_id_batch, show_skeleton=visualize_skeleton)
+
+    def calculateOKS(self, epochs, genEnum):
+        gen = self._get_generator(genEnum)
+        image_ids, list_of_predictions = self._full_list_of_predictions(gen, self.model_sub_dir, self.epoch)
+        oks = self.eval.oks_eval(image_ids, list_of_predictions, self.cocoGt)
+        print(oks)
+        return oks
+
+    def calculatePCK(self, epochs, genEnum):
+        gen = self._get_generator(genEnum)
+        _, list_of_predictions = self._full_list_of_predictions(gen, self.model_sub_dir, self.epoch)
+        pck = self.eval.pck_eval(list_of_predictions)
+        return pck
+
+    def plotOKS(self, epochs):
+        pass
+
+    def plotPCK(self, epochs):
+        pass
+
+    @lru_cache(maxsize=50)
+    def _full_list_of_predictions(self, gen, model_sub_dir, epoch):
```

Comment:
is this supposed to take `model_sub_dir, epoch`?

Reply:
It does this to make calls to it unique, this way the cache can be used

Reply:
ahh makes sense.